src/app/modules/item/pages/item-forum/item-forum.component.ts

```diff
@@ -79,12 +81,31 @@ export class ItemForumComponent implements OnInit, OnChanges, OnDestroy {
     ),
   );
 
+  state$ = combineLatest([
+    this.fetchState$,
+    this.nextRowIndexOpenThread$,
+  ]).pipe(
+    map(([ fetchState, nextRowIndexOpenThread ]) =>
+      ({
+        ...fetchState,
+        data: {
+          ...fetchState.data,
+          rowData: fetchState.data?.rowData.map((thread, index) => ({
+            ...thread,
+            rowIndex: index,
+            isOpened: index === nextRowIndexOpenThread,
```

Comment:
Then, to come back to your actual code... you store the row index and whether is is opened. 
Now I get what `nextRowIndexOpenThread` even I don't get its name at all (next what?).

As a first impression (that might be wrong), storing what row is selected looks strange to me while you could just ask to the discussion service what it is displaying. So you cause duplication of information, which typically causes inconsistency... ok that's easy to demo it: I click "open" (so the button changes to "close"),  I change tab, I come back on "Forum", the button is "Open" while the matching thread is open. 

There was maybe a reason to do that... because when going on a task, forum tab, there was a long list of thread (the same duplicated several time) due to a backend bug (not fixed)... and maybe you wanted one row to change to "close" ?

Anyway, I think the only good approach is just to display "open" or "close" just depending on whether the currently displayed discussion in the one of the row. And that simplifies all your code ts+template.